plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/PositionDeleteFilter.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import io.airlift.slice.Slice;
+import io.trino.plugin.iceberg.IcebergColumnHandle;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.connector.ConnectorPageSource;
+import org.apache.iceberg.MetadataColumns;
+import org.apache.iceberg.Schema;
+import org.roaringbitmap.longlong.ImmutableLongBitmapDataProvider;
+import org.roaringbitmap.longlong.LongBitmapDataProvider;
+import org.roaringbitmap.longlong.Roaring64Bitmap;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public final class PositionDeleteFilter
+        implements DeleteFilter
+{
+    private final ImmutableLongBitmapDataProvider deletedRows;
+
+    private PositionDeleteFilter(ImmutableLongBitmapDataProvider deletedRows)
+    {
+        this.deletedRows = requireNonNull(deletedRows, "deletedRows is null");
+    }
+
+    @Override
+    public Schema schema()
+    {
+        return new Schema(MetadataColumns.ROW_POSITION);
+    }
+
+    @Override
+    public RowPredicate createPredicate(List<IcebergColumnHandle> columns)
+    {
+        int filePosChannel = rowPositionChannel(columns);
+        return (page, position) -> {
+            long filePos = BIGINT.getLong(page.getBlock(filePosChannel), position);
+            return !deletedRows.contains(filePos);
+        };
+    }
+
+    private static int rowPositionChannel(List<IcebergColumnHandle> columns)
+    {
+        for (int i = 0; i < columns.size(); i++) {
+            if (columns.get(i).isRowPositionColumn()) {
+                return i;
```

Comment:
Verify only one match.

Reply:
We don't add multiple row position columns, but if we did, it wouldn't matter. Any of them would do as they would all have the same data.